src/devices/bus/mc10/mc10cart.cpp

```diff
@@ -0,0 +1,239 @@
+
+    MC-10 / Alice expansion slot - typically used for the 16K external
+    RAM expansion.
+
+         MC-10 and Alice pinout listing
+       ---  -------           ---  -------
+         1  GND                18  A6
+         2  GND                19  A7
+         3  D0                 20  A8
+         4  D1                 21  A9
+         5  D2                 22  A10
+         6  D3                 23  A11
+         7  D4                 24  A12
+         8  D5                 25  A13
+         9  D6                 26  A14
+        10  D7                 27  A15
+        11  R/!W               28  E
+        12  A0                 29  SEL
+        13  A1                 30  RESET
+        14  A2                 31  !NMI
+        15  A3                 32  +5V
+        16  A3                 33  GND
+        17  A5                 34  GND
+
+	SEL is an input to the MC-10 that allows the cartridge to remove
+	the internal chips from the bus.
+
+*********************************************************************/
+
+#include "emu.h"
+#include "mc10cart.h"
+
+#include "mc10_mcx128.h"
+#include "mc10_pak.h"
+#include "mc10_ram.h"
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(MC10CART_SLOT, mc10cart_slot_device, "mc10cart_slot", "MC-10 Cartridge Slot")
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+//-------------------------------------------------
+//  mc10cart_slot_device - constructor
+//-------------------------------------------------
+mc10cart_slot_device::mc10cart_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock) :
+	device_t(mconfig, MC10CART_SLOT, tag, owner, clock),
+	device_single_card_slot_interface<device_mc10cart_interface>(mconfig, *this),
+	device_image_interface(mconfig, *this),
+	m_nmi_callback(*this),
+	m_cart(nullptr),
+	m_memspace(*this, finder_base::DUMMY_TAG, -1)
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void mc10cart_slot_device::device_start()
+{
+	m_nmi_callback.resolve();
+	m_cart = get_card_device();
+}
+
+//-------------------------------------------------
+//  install_bank - install RAM in host computer
+//-------------------------------------------------
+
+void mc10cart_slot_device::install_bank(offs_t start, offs_t end, uint8_t *data)
+{
+	m_memspace->install_ram(start, end, data);
+}
+
+//-------------------------------------------------
+//  install_bank - install ROM in host computer
+//-------------------------------------------------
+
+void mc10cart_slot_device::install_rom(offs_t start, offs_t end, uint8_t *data)
+{
+	m_memspace->install_rom(start, end, data);
+}
```

Comment:
Is there really much use for these?  If the slot provides the cartridge with an accessor for the memory space, it can install itself with more flexibility.  These seem to be trampolines for the sake of it.